gRPC 1.30.0 doesn't build with LLVM 8.0.0 and 9.0.0

<!--
PLEASE DO NOT POST A QUESTION HERE.
This form is for bug reports and feature requests ONLY!

For general questions and troubleshooting, please ask/look for answers at StackOverflow, with "grpc" tag: https://stackoverflow.com/questions/tagged/grpc

For questions that specifically need to be answered by gRPC team members, please ask/look for answers at grpc.io mailing list: https://groups.google.com/forum/#!forum/grpc-io

Issues specific to *grpc-java*, *grpc-go*, *grpc-node*, *grpc-dart*, *grpc-web* should be created in the repository they belong to (e.g. https://github.com/grpc/grpc-LANGUAGE/issues/new)
-->

### What version of gRPC and what language are you using?
1.30.0, C++ (core is failing to build so the language is probably irrelevant?)


### What operating system (Linux, Windows,...) and version?
Linux (Archlinux)

### What runtime / compiler are you using (e.g. python version or version of gcc)
Building with Bazel and LLVM - tried both 8.0.0 and 9.0.0 (using this Bazel toolchain: https://github.com/grailbio/bazel-toolchain)

### What did you do?
Just tried to build gRPC - `bazel build @com_github_grpc_grpc//:grpc` in a repo fetching gRPC 1.30.0 without any patch.

### What did you expect to see?
A successful build.

### What did you see instead?
```
ERROR: /home/kernald/.cache/bazel/_bazel_kernald/551cfa9086d3d42cb8d718a9a97333ef/external/com_github_grpc_grpc/BUILD:2111:16: C++ compilation of rule '@com_github_grpc_grpc//:tsi' failed (Exit 1) clang failed: error executing command external/llvm_toolchain/bin/clang -U_FORTIFY_SOURCE -fstack-protector -fno-omit-frame-pointer -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign '-std=c++17' '-stdlib=libc++' -MD -MF ... (remaining 50 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
external/com_github_grpc_grpc/src/core/tsi/ssl_transport_security.cc:1696:13: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
            std::string(name_subdomain).c_str());
            ^
external/llvm_toolchain/bin/../include/c++/v1/iosfwd:210:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
1 error generated.
Target @com_github_grpc_grpc//:grpc failed to build
```

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

### Anything else we should know about your project / environment?

--- a/src/core/tsi/ssl_transport_security.cc
+++ b/src/core/tsi/ssl_transport_security.cc
@@ -33,6 +33,8 @@
 #include <sys/socket.h>
 #endif
 
+#include <string>
+
 #include <grpc/grpc_security.h>
 #include <grpc/support/alloc.h>
 #include <grpc/support/log.h>

